tests/index.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { mount, ReactWrapper, StatelessComponent } from 'enzyme';
+
+import { progressivelyEnhance, withIsEnhanced } from '../src/consumer';
+import { ProgressiveEnhancementProp } from '../src/context';
+import { enableProgressiveEnhancementsOnMount } from '../src/provider';
+
+const TestComponent: React.SFC<{ name: string }> = ({ name }) => <div>{name}</div>;
+const Loading: React.SFC<{}> = () => <div> Loading!!! </div>;
```

Comment:
I like the idea behind the loading option, but I have mixed feelings about the name.

The fundamental principle behind the progressive enhancement idea is resilience—the enhancement may fail to load for many reasons, and that's OK. But if it does, we wouldn't want to show an infinite loading spinner…

Perhaps a better name would be `BeforeEnhanced`, `Unenhanced`, or similar? WDYT?

Can you think of any good examples where you would use this? Maybe that will help us frame our thinking re. naming.

Reply:
Hmm, that's fair actually. I felt like it could be helpful for others, but I don't see us needing to ever use it. I'd rather simply remove it from the codebase, and re-introduce it once we see a need for it. 👍